File: BricksGDX/core/src/com/salvador/bricks/game_objects/MenuText.java
package com.salvador.bricks.Objects;

import static com.salvador.bricks.Objects.ResourceManager.getFont;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.GlyphLayout;
import com.badlogic.gdx.scenes.scene2d.Actor;

public class MenuText extends Actor {

    BitmapFont font;
    float textWidth;
    float x, y;
    GlyphLayout layout;
    private String text;

    public MenuText(String text, String fontName, float x, float y) {

        this.x = x;
        this.y = y;
        this.text = text;

        font = getFont(fontName);
        layout = new GlyphLayout();
        layout.setText(font, text);
        textWidth = layout.width;
    }

    @Override
    public void act(float delta) {
        super.act(delta);
        layout.setText(font, text);
        textWidth = layout.width;
    }

    public void setText(String text) {
        this.text = text;
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        font.draw(batch, text, 450F / 2 - textWidth / 2, y);
    }
}

File: BricksGDX/core/src/com/salvador/bricks/game_objects/MenuText.kt
package com.salvador.bricks.game_objects

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.graphics.g2d.GlyphLayout
import com.badlogic.gdx.scenes.scene2d.Actor

class MenuText(@JvmField var text: String, fontName: String, var textY: Float) : Actor() {

    var font: BitmapFont = ResourceManager.getFont(fontName)
    var textWidth: Float
    var layout = GlyphLayout()

    init {
        layout.setText(font, text)
        textWidth = layout.width
    }

    override fun act(delta: Float) {
        super.act(delta)
        layout.setText(font, text)
        textWidth = layout.width
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        font.draw(batch, text, 450F / 2 - textWidth / 2, textY)
    }
}
